### PR TITLE
Add add support for load dataset from DataBroker

### PR DESCRIPTION
**Pull request by cjh1**
This adds the ability to load data from databroker. The functionality is disabled if tomviz is running in a environment without the databroker package.

### CODE REVIEW

**Comment by cjh1** [comment]
@cryos This adds the basic databroker load functionality. There are currently two issue that I may need some assistants with from someone a little more familiar with databroker.

- If we list out the variables in a dataset and then close tomviz, the following error is printed:

```
HDF5: infinite loop closing library
      L,T_top,P,P,Z,FD,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E,E
```

- Once a dataset is loaded tomviz will no longer exit. Something is prevent the interpreter for exiting ( I haven't looked further ) 

I think that both of these are related to resource cleanup in databroker. I have tried closing the datasets but that didn't seem to help. Are there other cleanup methods I need to call?

File: tomviz/python/tomviz/io/_databroker.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    finally:
+        dataset.close()
+
+    variables = sorted(variables, key=lambda v: v['name'])
+
+    return variables
+
+
+def load_variable(catalog_name, run_uid, table, variable):
+    data = catalog[catalog_name][run_uid][table].read()[variable].data
```
[the remark this answers is not in the document]
Comment (cjh1) [reply tomviz/python/tomviz/io/_databroker.py:79]: Good catch

File: tomviz/Variant.cxx [hunk 2]
```diff
@@ -35,11 +35,21 @@ Variant::Variant(const vector<Variant>& l) : m_type(LIST)
   new (&m_value.listVal) vector<Variant>(l);
 }
 
+Variant::Variant(const map<string, Variant>& m) : m_type(MAP)
+{
+  new (&m_value.mapVal) map<string, Variant>(m);
```
[the remark this answers is not in the document]
Comment (cjh1) [reply tomviz/Variant.cxx:40]: yep!

File: tomviz/Utilities.cxx [hunk 3]
```diff
@@ -771,6 +771,8 @@ Variant toVariant(const QVariant& value)
       QVariantList list = value.toList();
       return toVariant(list);
     }
+    case QVariant::Map: {
+    }
```
[the remark this answers is not in the document]
Comment (cjh1) [reply tomviz/Utilities.cxx:775]: Oops!

File: tomviz/DataBrokerLoadReaction.cxx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  DataBrokerLoadDialog dialog(dataBroker, tomviz::mainWidget());
+
+  if (dialog.exec() == QDialog::Accepted) {
+    auto catalog = dialog.selectedCatalog();
+    auto runUid = dialog.selectedRunUid();
+    auto table = dialog.selectedTable();
+    auto variable = dialog.selectedVariable();
+
+    tomviz::mainWidget()->setCursor(Qt::WaitCursor);
+    auto call = dataBroker->loadVariable(catalog, runUid, table, variable);
```
[the remark this answers is not in the document]
Comment (cjh1) [reply tomviz/DataBrokerLoadReaction.cxx:41]: Its a client of `dataBroker` which is deleted in the lambda so will get cleaned up.

File: tomviz/MainWindow.cxx [hunk 5]
```diff
@@ -539,6 +543,11 @@ MainWindow::MainWindow(QWidget* parent, Qt::WindowFlags flags)
             m_ui->actionAcquisition->setEnabled(true);
             m_ui->actionPassiveAcquisition->setEnabled(true);
             registerCustomOperators(pythonWatcher->result());
+            // Check if we have DataBroker and enable menu if we do
+            auto dataBroker = new DataBroker(this);
```
[the remark this answers is not in the document]
Comment (cjh1) [reply tomviz/MainWindow.cxx:547]: Will clean it up.

**Comment by cjh1** [comment]
@psavery Done

File: tomviz/DataBrokerLoadReaction.cxx (hunk 4 above)
[the remark this answers is not in the document]
Comment (psavery) [reply tomviz/DataBrokerLoadReaction.cxx:41]: I see, makes sense.

**Review by psavery — APPROVED** [review]
LGTM

**Review by cryos — COMMENTED** [review]
I need to test it, but looking through the code this looks like a nice way of structuring with a solid Python layer that can integrate any minor updates to the interaction with Data Broker.

**Pull request merged**